src/widget/woverviewrgb.cpp

```diff
@@ -71,24 +84,36 @@ bool WOverviewRGB::drawNextPixmapPart() {
         unsigned char left = pWaveform->getAll(currentCompletion);
         unsigned char right = pWaveform->getAll(currentCompletion + 1);
 
-        low = pWaveform->getLow(currentCompletion);
-        mid = pWaveform->getMid(currentCompletion);
-        high = pWaveform->getHigh(currentCompletion);
+        low  = (qreal) pWaveform->getLow(currentCompletion);
+        mid  = (qreal) pWaveform->getMid(currentCompletion);
+        high = (qreal) pWaveform->getHigh(currentCompletion);
+
+        // Do matrix multiplication
+        red   = low * lowColor_r + mid * midColor_r + high * highColor_r;
+        green = low * lowColor_g + mid * midColor_g + high * highColor_g;
+        blue  = low * lowColor_b + mid * midColor_b + high * highColor_b;
 
-        max = (float) math_max3(low, mid, high);
+        // Normalize and draw
+        max = math_max3(red, green, blue);
         if (max > 0.0f) {
```

Comment:
since max is no longer a float, remove the f suffix which forces the literal to a float value (instead of double)